gui/functionality/add_cluster.py

```diff
@@ -82,18 +98,32 @@ def add_cluster(self):
             setup_actions = SetupActionsConfigImpl()
             setup_actions.jupyter = parser.parse_actions(self.actions_file_name)
 
-        if auth == 'PUBLIC_KEY':
-            key = self.ui.key_type_box.currentText()
-            if key == 'RSA_KEY':
-                cluster = add_cluster(name=cluster_name,
-                                      user=user,
-                                      host=host,
-                                      port=port,
-                                      auth=AuthMethod.PUBLIC_KEY,
-                                      key=KeyType.RSA,
-                                      install_key=True,
-                                      setup_actions=setup_actions,
-                                      use_jupyter_lab=use_jupyter_lab)
+        if auth == 'GENERATE_NEW_KEY':
+            cluster = add_cluster(name=cluster_name,
+                                  user=user,
+                                  host=host,
+                                  port=port,
+                                  auth=AuthMethod.PUBLIC_KEY,
+                                  key=KeyType.RSA,
+                                  install_key=True,
+                                  setup_actions=setup_actions,
+                                  use_jupyter_lab=use_jupyter_lab)
+            node = cluster.get_access_node()
+            node.connect(password=password)
+            save_environment()
+        elif auth == 'PRIVATE_KEY':
+            cluster = add_cluster(name=cluster_name,
+                                  user=user,
+                                  host=host,
+                                  port=port,
+                                  auth=AuthMethod.PRIVATE_KEY,
+                                  key=self.key_name,
```

Comment:
The `key` argument is of a `KeyType` type(currently only KeyType.RSA is supported) you shouldn't put a name in here.